staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -172,35 +172,38 @@ func getExcludedChecks(r *http.Request) sets.String {
 // handleRootHealthz returns an http.HandlerFunc that serves the provided checks.
 func handleRootHealthz(checks ...HealthChecker) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		failed := false
 		excluded := getExcludedChecks(r)
-		var verboseOut bytes.Buffer
+		// failedVerboseLogOutput is for output to the log.  It indicates detailed failed output information for the log.
+		var failedVerboseLogOutput bytes.Buffer
+		var failedChecks []string
+		var individualCheckOoutput bytes.Buffer
 		for _, check := range checks {
 			// no-op the check if we've specified we want to exclude the check
 			if excluded.Has(check.Name()) {
 				excluded.Delete(check.Name())
-				fmt.Fprintf(&verboseOut, "[+]%v excluded: ok\n", check.Name())
+				fmt.Fprintf(&individualCheckOoutput, "[+]%v excluded: ok\n", check.Name())
 				continue
 			}
 			if err := check.Check(r); err != nil {
 				// don't include the error since this endpoint is public.  If someone wants more detail
 				// they should have explicit permission to the detailed checks.
-				klog.V(4).Infof("healthz check %v failed: %v", check.Name(), err)
-				fmt.Fprintf(&verboseOut, "[-]%v failed: reason withheld\n", check.Name())
-				failed = true
+				fmt.Fprintf(&individualCheckOoutput, "[-]%v failed: reason withheld\n", check.Name())
+				// but we do want detailed information for our log
+				fmt.Fprintf(&failedVerboseLogOutput, "[-]%v failed: %v\n", check.Name(), err)
```

Comment:
Slight paranoia check. I like the change but I note we are now logging the errors at V(2) where before we were logging them at V(4). If there is something sensitive in them, that data will now come out in the logs with no option to log only the failure, which use to be possible. I wonder if it would be worth having different sensitivity outputs based on the log level? I tend to think its the wrong tool for the job but without it there is no tool.

Reply:
We control the content of the error messages in code in this repo. It's also the same log that the failures these poststarthooks are experiencing are directly logging to already.  Is there some heightened concern about this log message versus the messages those bits of code are already writing?

Reply:
> Slight paranoia check. I like the change but I note we are now logging the errors at V(2) where before we were logging them at V(4). If there is something sensitive in them, that data will now come out in the logs with no option to log only the failure, which use to be possible. I wonder if it would be worth having different sensitivity outputs based on the log level? I tend to think its the wrong tool for the job but without it there is no tool.

Controller and post-start hooks should already be placing their errors at either Error or V((2) depending on severity, so related messages should already be in the log.  In cases where a message isn't already in the log and it is sensitive, the bug should be fixed and in the short term, you can immediately mitigate by excluding it from the health check while the patch is merged.

Reply:
I'll buy that as a short term work-around to address my concern.

Reply:
/lgtm